server/map_management.c

```diff
@@ -53,6 +54,107 @@ void		        manage_bombs(t_game_info *game_info)
   printf("\nmanage_bomb done\n");
 }
 
+int					bomb_kill_player(int x, int y) {
+	t_game_info 	*game_info;
+	int				z;
+	int				i;
+	SDL_Rect 		*walls;
+
+	game_info = get_game_info();
+	const SDL_Rect bomb = {x, y, PIXEL_SIZE, PIXEL_SIZE};
+	walls = get_walls();
+
+//	Check walls
+	for (i = 0; i < 82; i++) {
+		if (SDL_HasIntersection(&walls[i], &bomb)) {
+			printf("\nWALL at %d:%d i=%d\n", walls[i].x, walls[i].y, i);
+			printf("bomb explode on WALL\n\n");
+			return 1;
+		}
+	}
+
+//	Check destroyable map objects
+	for (i = 0; i < INLINE_MATRIX; ++i) {
+		if (!game_info->map_destroyable[i].exist && !game_info->map_destroyable[i].bomb)
+			continue;
+//		The x and y of the object are in fact their center, so we need to determine their real positions
+		const SDL_Rect element = {((int)(game_info->map_destroyable[i].x/48) + 1) * PIXEL_SIZE, ((int)(game_info->map_destroyable[i].y/48) + 1) * PIXEL_SIZE, PIXEL_SIZE, PIXEL_SIZE};
+		if (SDL_HasIntersection(&bomb, &element)) {
+			game_info->map_destroyable[i].dying = 1;
+			game_info->map_destroyable[i].exist = 0;
+			printf("bomb explode on wall destroyable\n\n");
+			return 1;
+		}
+	}
+
+	// Loop over players
+	for (z = 0; z < (int)(sizeof(game_info->players)/sizeof(game_info->players[0])); ++z) {
+		if (&game_info->players[z] != NULL && &game_info->players[z].alive != 0)
+		{
+			//printf("\nin players stuff\n");
+			const SDL_Rect player = {game_info->players[z].x, game_info->players[z].y, PIXEL_SIZE, PIXEL_SIZE};
+			if (SDL_HasIntersection(&bomb, &player))
+			{
+				game_info->players[z].alive = 0;
+				game_info->players[z].dying++;
+			}
+		}
+	}
+
+	return 0;
+}
+
+void					bomb_check_players(int i) {
+	t_game_info 		*game_info;
+	t_map_destroyable 	*map_destroyable;
+	int					y;
+	int					x;
+	int					bomb_x;
+	int					bomb_y;
+	int					is_wall;
+
+	game_info = get_game_info();
+	map_destroyable = game_info->map_destroyable;
+//		The x and y of the object are in fact their center, so we need to determine their real positions
+	bomb_x = ((int)(map_destroyable[i].x/48) + 1) * PIXEL_SIZE;
+	bomb_y = ((int)(map_destroyable[i].y/45) + 1) * PIXEL_SIZE;
```

Comment:
non, ce calcul part d'une mauvaise hypothèse.
le x et le y représente le coin en haut à gauche de la zone, pas le centre. Du coup, ca donne des résultats très bizarres à l'écran.
En plus, si on sait placer au pixel, pour ce type de vérif je pense qu'on peut se baser sur les cases. Si ton bomber est sur la case de l'explosion ca pète...

Reply:
C'était un calcul qui se base sur le constat que le x et y sont impair alors qu'on est censé faire des cases de 48x48.

Reply:
la logique c'est : 
x ou y sont impairs donc ils indiquent le centre, donc je fais x/48 +1 * PIXEL_SIZE et y/45(?!) + 1 * PIXEL_SIZE
???
Pour moi il y a deux choses:
 - on place précisément les objets grâce à x et y qui fonctionnent aux pixels et définissent le coin en haut à gauche des SDL_Rect que l'on place
 - pour pas itérer pixel par pixel, on travaille avec des cases: la map destroyable ou la fixed map (qu'on ne stocke nulle part d'ailleurs ce qui est un peu con)

Donc quand tu as x = 333 et y = 444
tu es sur la case  (x + y * COLUMNS) / PIXEL_SIZE;
soit pour l'exemple: (333 + 444 * 15) / 48
soit map_destroyable[145] (tronqué).
Ce calcul est nécessaire car la map est inline.

Au final, je vois pas le rapport entre x || y == impair et dimension === centre...